back/app/models/initiative.rb

```diff
@@ -96,9 +96,10 @@ class Initiative < ApplicationRecord
   end)
 
   scope :with_status_code, (proc do |code|
-    joins('LEFT OUTER JOIN initiative_initiative_statuses ON initiatives.id = initiative_initiative_statuses.initiative_id')
-      .joins('LEFT OUTER JOIN initiative_statuses ON initiative_statuses.id = initiative_initiative_statuses.initiative_status_id')
-      .where(initiative_statuses: { code: code })
+    left_outer_joins(:initiative_initiative_status)
+      .where(initiative_initiative_statuses: {
+        initiative_status_id: InitiativeStatus.where(code: code).select(:id)
```

Comment:
Does it work?
```suggestion
        initiative_statuses: { code: code }
```
I'm happy to have a call to test and discuss it. But in general, I think I like the new implementation more than the old one.

Reply:
No, this change doesn't work...
```
# PG::UndefinedTable:
#   ERROR:  missing FROM-clause entry for table "initiative_statuses"
#   LINE 1: ...nitiatives"."publication_status" IN ($1, $2) AND ("initiativ...
```
But the smaller [change you recommended](https://github.com/CitizenLabDotCo/citizenlab/pull/5444#discussion_r1272190824) does work.